packages/react-tooltip/src/components/Tooltip/useTooltip.tsx

```diff
@@ -30,52 +45,117 @@ export const useTooltip = (
   ref: React.Ref<HTMLElement>,
   defaultProps?: TooltipProps,
 ): TooltipState => {
-  const [target, setTarget] = React.useState<HTMLElement | null>(null);
+  const [visible, setVisible] = React.useState(false);
 
   const state = mergeProps(
     {
       ref,
-      as: 'div',
+      children: props.children,
+      content: {
+        as: React.Fragment,
+      },
+      id: useId('tooltip-'),
       role: 'tooltip',
       position: 'above',
       align: 'center',
       offset: 4,
-      arrow: { as: 'div', children: '' },
-      visible: !!target,
+      showDelay: 250,
+      hideDelay: 250,
+      visible,
+      rendered: !useIsSSR(),
     },
     defaultProps && resolveShorthandProps(defaultProps, tooltipShorthandProps),
     resolveShorthandProps(props, tooltipShorthandProps),
   );
 
-  if (state.noArrow) {
-    state.arrow = undefined;
-  }
-
-  React.useImperativeHandle(
-    state.componentRef,
-    () => ({
-      show: t => setTarget(t),
-      hide: () => setTarget(null),
-    }),
-    [],
-  );
-
   const theme = useTheme();
 
-  const { targetRef, containerRef, arrowRef } = usePopper({
+  const popper = usePopper({
     enabled: state.visible,
     position: state.position,
     align: state.align,
     offset: [0, state.offset + (state.noArrow ? 0 : arrowHeight)],
     arrowPadding: theme?.global ? 2 * parseInt(tooltipBorderRadius(theme), 10) : 0,
   });
 
-  targetRef.current = target;
-  state.ref = useMergedRefs(state.ref, containerRef);
+  state.ref = useMergedRefs(state.ref, popper.containerRef);
+  state.arrowRef = popper.arrowRef;
+
+  const tooltipManager = useTooltipManager(createTooltipManager);
+
+  // Notify the manager when the pointer enters or leaves the tooltip
+  state.onPointerEnter = mergeCallbacks(() => tooltipManager.notifyEnterTooltip(), state.onPointerEnter);
+  state.onPointerLeave = mergeCallbacks(() => tooltipManager.notifyLeaveTooltip(), state.onPointerLeave);
+
+  const onEnterTrigger = (ev: React.SyntheticEvent<HTMLElement>) => {
+    const target = state.targetRef?.current ?? ev.currentTarget;
```

Comment:
should a user defined `target` prop be supported too ?

Reply:
It's supported by setting the `targetRef` prop. Although I realize that could be confusing since refs are usually "out" props (set by the component), rather than "in" props. 

I could change the prop to just be `target` (not a ref), but that would require the tooltip to be re-rendered after the target element is rendered.

Reply:
`refs` are generally a bad kind of prop to accept since mutating them does not result in rerenders and you can't use them as dependencies in effects or memos... 

That being said I think I'll need to take another look to see how the `targetRef` might be replaced with a straight up DOM node.